python/lsst/pipe/supertask/config.py

```diff
@@ -47,6 +49,66 @@
 # ------------------------
 
 
+def makeDatasetField(name, dtype, removeUnitsVariable=False):
+    def factory(**kwargs):
+        return pexConfig.ConfigField(doc=kwargs['doc'],
+                                     dtype=dtype,
+                                     default=dtype(
+                                         **{k: v for k, v in kwargs.items()
+                                             if k not in ('doc', 'check')}),
+                                     check=kwargs['check'])
+    if removeUnitsVariable:
+        def wrappedFunc(doc, name, storageClass, check=None):
+            return factory(**{k: v for k, v in locals().items() if k != 'factory'})
+        unitsDoc = ""
+        unitsString = ""
+    else:
+        def wrappedFunc(doc, name, uints, storageClass, check=None):
+            return factory(**{k: v for k, v in locals().items() if k != 'factory'})
+        unitsDoc = """
+        units : iterable of `str`
+        Iterable of DataUnits for this `~lsst.daf.butler.DatasetType`
+        """
+        unitsString = """,
```

Comment:
This line looks a bit odd with the new line. Why isn't this:
```python
unitString = "units, "
```
? A comment would help.